src/workload.rs

```diff
@@ -118,51 +181,40 @@ impl fmt::Display for Upstream {
     }
 }
 
-fn byte_to_ip(b: bytes::Bytes) -> Result<Option<IpAddr>, WorkloadError> {
-    Ok(match b.len() {
-        0 => None,
-        4 => {
-            let v: [u8; 4] = (&*b).try_into().expect("size already proven");
-            Some(IpAddr::V4(Ipv4Addr::from(v)))
-        }
-        16 => {
-            let v: [u8; 16] = (&*b).try_into().expect("size already proven");
-            Some(IpAddr::V6(Ipv6Addr::from(v)))
-        }
-        n => return Err(WorkloadError::ByteAddressParse(n)),
-    })
+fn byte_to_ip(b: &bytes::Bytes) -> Result<Option<IpAddr>, WorkloadError> {
+    let s = std::str::from_utf8(b).expect("error converting bytes to string");
+    if s.is_empty() {
+        return Ok(None);
+    }
+    match IpAddr::from_str(s) {
```

Comment:
Is this doing allocations to convert bytes -> string -> ip instead of just bytes -> IP?

I expect IP addresses to be one of the largest memory usages since we want to scale to >100k IPs, so optimizing this is pretty important

Reply:
I am hoping to add benchmarks so we don't have to guess at this :-)